tests/urereduct/rules/ReductRulesUTest.cxxtest

```diff
@@ -0,0 +1,212 @@
+/*
+ * tests/pln/rules/ReductRulesUTest.cxxtest
+ *
+ * Copyright (C) 2019 Kasim
+ * All Rights Reserved
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License v3 as
+ * published by the Free Software Foundation and including the exceptions
+ * at http://opencog.org/wiki/Licenses
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program; if not, write to:
+ * Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include <opencog/guile/SchemeEval.h>
+#include <opencog/guile/SchemeSmob.h>
+#include <opencog/atomspace/AtomSpace.h>
+#include <cxxtest/TestSuite.h>
+#include <string>
+
+using namespace opencog;
+using namespace std;
+
+class ReductRulesUTest : public CxxTest::TestSuite
+{
+private:
+	AtomSpace as;
+	SchemeEval eval;
+
+public:
+	ReductRulesUTest() : eval(&as)
+	{
+		logger().set_level(Logger::DEBUG);
+		logger().set_timestamp_flag(false);
+		logger().set_print_to_stdout_flag(true);
+
+		const string reductUREUTest_rules_dir=string(PROJECT_SOURCE_DIR) + "/tests/urereduct/rules";
+		const string reductURE_rules_dir=string(PROJECT_SOURCE_DIR) + "/opencog/urereduct/rules";
+
+		eval.eval(string("(add-to-load-path \"") + reductUREUTest_rules_dir + string("\")"));
+		eval.eval(string("(add-to-load-path \"") + reductURE_rules_dir + string("\")"));
+
+		eval.eval("(use-modules (opencog))");
+		eval.eval("(use-modules (opencog exec) (opencog reduct))");
+	}
+
+	~ReductRulesUTest()
+	{
+		// Erase the log file if no assertions failed.
+		if (!CxxTest::TestTracker::tracker().suiteFailed())
+			std::remove(logger().get_filename().c_str());
+	}
+
+	void setUp();
+
+	void tearDown();
+
+	void load_scm_files(std::vector<std::string>);
+
+	void test_and_idenityT();
+    void test_or_identityT();
+    void test_and_identityF();
+    void test_or_identityF();
+    void test_or_identity();
```

Comment:
Cool!

@Eman22S, can you just fix the indentation (tab or smart tab, rather than space)? Thanks.